homeassistant/components/ness_alarm/__init__.py

```diff
@@ -62,101 +66,212 @@
-        """Receives and propagates zone state updates."""
-        async_dispatcher_send(
-            hass, SIGNAL_ZONE_CHANGED, ZoneChangedData(zone_id=zone_id, state=state)
-        )
+    await async_setup_services(hass)
 
-    def on_state_change(arming_state: ArmingState, arming_mode: ArmingMode | None):
-        """Receives and propagates arming state updates."""
-        async_dispatcher_send(
-            hass, SIGNAL_ARMING_STATE_CHANGED, arming_state, arming_mode
-        )
+    return True
 
-    client.on_zone_change(on_zone_change)
-    client.on_state_change(on_state_change)
 
-    async def handle_panic(call: ServiceCall) -> None:
-        await client.panic(call.data[ATTR_CODE])
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+    if unload_ok:
+        data = hass.data[DOMAIN].pop(entry.entry_id)
+        await data["client"].close()
+
+    return unload_ok
+
+
+async def update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle options update."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
+async def async_setup_services(hass: HomeAssistant) -> None:
+    """Set up services for Ness Alarm."""
+    if hass.services.has_service(DOMAIN, SERVICE_AUX):
+        return
 
     async def handle_aux(call: ServiceCall) -> None:
-        await client.aux(call.data[ATTR_OUTPUT_ID], call.data[ATTR_STATE])
+        """Handle aux service call."""
+        output_id = call.data.get("output_id")
+        state = call.data.get("state")
+
+        for entry_data in hass.data[DOMAIN].values():
+            if isinstance(entry_data, dict) and "client" in entry_data:
+                client = entry_data.get("client")
+                if client:
+                    await client.aux(output_id, state)
+
+    async def handle_panic(call: ServiceCall) -> None:
+        """Handle panic service call."""
+        code = call.data.get(SERVICE_CODE)
+
+        for entry_data in hass.data[DOMAIN].values():
+            if isinstance(entry_data, dict) and "client" in entry_data:
+                client = entry_data.get("client")
+                if client:
+                    await client.panic(code)
 
     hass.services.async_register(
-        DOMAIN, SERVICE_PANIC, handle_panic, schema=SERVICE_SCHEMA_PANIC
+        DOMAIN,
+        SERVICE_AUX,
+        handle_aux,
+        schema=vol.Schema(
+            {
+                vol.Required("output_id"): cv.positive_int,
+                vol.Required("state", default=True): cv.boolean,
+            }
+        ),
     )
+
     hass.services.async_register(
-        DOMAIN, SERVICE_AUX, handle_aux, schema=SERVICE_SCHEMA_AUX
+        DOMAIN,
+        SERVICE_PANIC,
+        handle_panic,
+        schema=vol.Schema(
+            {
+                vol.Optional(SERVICE_CODE): cv.string,
+            }
+        ),
     )
```

Comment:
Okay so the current issue with these services are is that they execute for _all_ entries. As this was only configurable once, we should mark it as single_config_entry: true in the manifest to avoid collisions. In the future we should expand the selector to select what system we target, and then we can allow multiple entries afterwards.

Also, we should register these in `async_setup`